docs/glossary.md

```diff
@@ -0,0 +1,129 @@
+
+The ASAR format was created primarily to improve performance on Windows... TODO
+
+### Brightray
+
+Brightray is a static library that makes [libchromiumcontent] easier to use in
+applications. It was created specifically for Electron, but can be used to
+enable Chromium's renderer in native apps that are not based on Electron.
+
+Brightray is a low-level dependency of Electron that does not concern the
+majority of Electron users.
+
+### IPC
+
+IPC stands for Inter-Process Communication. Electron uses IPC to send
+serialized JSON messages between the [main] and [renderer] processes.
+
+### libchromiumcontent
+
+A single, shared library that includes the Chromium Content module and all its
+dependencies (e.g., Blink, [V8], etc.).
+
+### main process
+
+The main process, commonly a file named `main.js`, is the entry point to every
+Electron app. It controls the life of the app, from open to close. It also
+manages native elements such as the Menu, Menu Bar, Dock, Tray, etc.  The
+main process is responsible for creating each new renderer process in the app.
+The full Node API is built in.
+
+Every app's main process file is specified in the `main` property in
+`package.json`. This is how `electron .` knows what file to execute at startup.
+
+### MAS
+
+Acronym for Apple's Mac App Store. For details on submitting your app to the
+MAS, see the [Mac App Store Submission Guide].
+
+### native modules
+
+Native modules (also called [addons] in
+Node.js) are modules written in C or C++ that can be loaded into Node.js or
+Electron using the require() function, and used just as if they were an
+ordinary Node.js module. They are used primarily to provide an interface
+between JavaScript running in Node.js and C/C++ libraries.
+
+Native Node modules are supported by Electron, but since Electron is very
+likely to use a different V8 version from the Node binary installed in your
+system, you have to manually specify the location of Electron’s headers when
+building native modules.
+
+See also [Using Native Node Modules].
+
+### process
+
+A process is an instance of a computer program that is being executed. Electron
+apps that make use of the [main] and one or many [renderer] process are
+actually running several programs simultaneously.
+
+In Node.js and Electron, each running process has a `process` object. This
+object is a global that provides information about, and control over, the
+current process. As a global, it is always available to applications without
+using require().
+
+### renderer process
+
+The renderer process is a browser window in your app. Unlike the main process,
+there can be multiple of these and each is run in a separate process.
+They can also be hidden.
+
+In normal browsers, web pages usually run in a sandboxed environment and are not
+allowed access to native resources. Electron users, however, have the power to
+use Node.js APIs in web pages allowing lower level operating system
+interactions.
+
+### Squirrel
```

Comment:
Might be worth noting that there are two different implementation of Squirrel.

`Squirrel.Mac` and `Squirrel.Windows`